src/register.jl

```diff
@@ -51,17 +51,22 @@ macro register_symbolic(expr, define_promotion = true, Ts = [])
     eval_method = :(@eval function $f($(Expr(:$, :(s...))),)
                         args = [$(Expr(:$, :(s_syms...)))]
                         unwrapped_args = map($unwrap, args)
-                        if !any(x->$issym(x) || $istree(x), unwrapped_args)
+                        res = if !any(x->$issym(x) || $istree(x), unwrapped_args)
                             if typeof.(args) == typeof.(unwrapped_args)
                                 # this means there was no method which already caught it.
-                                # so create a term
-                                $Term{$ret_type}($f, unwrapped_args)
+                                # so rethrow method error
+                                throw(MethodError($f, unwrapped_args))
```

Comment:
Should this just be
```suggestion
                                $f(unwrapped_args...)
```
?